tests/auto/tst_logins/components/LoginsTestHelper.js

```diff
@@ -0,0 +1,83 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ * Copyright (c) 2021 Open Mobile Platform LLC.
+ */
+
+const Cc = Components.classes;
+const Ci = Components.interfaces;
+const Cu = Components.utils;
+
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+Cu.import("resource://gre/modules/Services.jsm");
+
+XPCOMUtils.defineLazyServiceGetter(Services, "embedlite",
+                                    "@mozilla.org/embedlite-app-service;1",
+                                    "nsIEmbedAppService");
+
+Services.scriptloader.loadSubScript("chrome://embedlite/content/Logger.js");
+
+const LoginInfo = Components.Constructor("@mozilla.org/login-manager/loginInfo;1",
+                                         "nsILoginInfo", "init");
+
+function LoginsTestHelper() {
+  Logger.debug("JSComp: LoginsTestHelper.js loaded");
+}
+
+LoginsTestHelper.prototype = {
+  classID: Components.ID("{417f9989-029d-4560-85c0-704973591d5d}"),
+
+  QueryInterface: XPCOMUtils.generateQI([Ci.nsIObserver]),
+
+  __pwmgr : null, // Password Manager service
+  get _pwmgr() {
+    if (!this.__pwmgr)
+      this.__pwmgr = Cc["@mozilla.org/login-manager;1"].
+                     getService(Ci.nsILoginManager);
+    return this.__pwmgr;
+  },
+
+  observe: function(aSubject, aTopic, aData) {
```

Comment:
I'm failing to understand need for this? Why are you not using the LoginsHelper.js of the embedlite-components? The embedlite-components part if essential bit of this whole thing and now there's some "extra" that we need to remember to fix always when we do changes.

If the issue is that "add" is missing from LoginsHelper.js, then I'd rather add that. Profile must be in case different from the normal browser use. This means that this is never tampering the passwords saved by the browser.

Reply:
As discussed elsewhere, this was just about keeping the `add()` and `removeAll()` actions out of the non-test codebase where they're not used. However, I agree that it'll be more succinct and easier to maintain if they're kept together in the same place.

To this end I've moved what was the content of LoginsTestHelper.js into LoginsHelper,js, so this now depends on this change to embedlite-components: https://github.com/sailfishos/embedlite-components/pull/3